File: examples/functions/rest-api-methods/products/post.lua
-- POST /products — create a product
local cjson = require("cjson")
local shared = require("_shared")

local function handler(event)
    local ok, data = pcall(cjson.decode, event.body or "")
    if not ok then
        return shared.json_response(400, { error = "Invalid JSON" })
    end

    local name = shared.trim_text(data.name or "")
    local price = data.price or 0

    if name == "" then
        return shared.json_response(400, { error = "name is required" })
    end

    return shared.json_response(201, { id = 42, name = name, price = price, created = true })
end

return handler

File: examples/functions/rest-api-methods/products/post.php
<?php
// POST /products — create a product
function handler($event) {
    $body = $event["body"] ?? "";
    $data = is_string($body) ? json_decode($body, true) : ($body ?: []);

    $name = trim($data["name"] ?? "");
    $price = $data["price"] ?? 0;

    if ($name === "") {
        return [
            "status" => 400,
            "body" => json_encode(["error" => "name is required"]),
        ];
    }

    return [
        "status" => 201,
        "headers" => ["Content-Type" => "application/json"],
        "body" => json_encode(["id" => 42, "name" => $name, "price" => $price, "created" => true]),
    ];
}
